### DISCUSSION TITLE
After having used Micropython 1.20 on ESP32, can't flash Tasmota anymore (on an Olimex ESP32-PoE)

### DISCUSSION THREAD

**Discussion by hagenbuch** [arendst/Tasmota#19695] [Q&A] [answered]
Hi all,

I've made some experiments with Micropython 1.20 on an Olimex ESP32-POE device.

Now I would like to flash Tasmota on it but neither the web installer not esptool.py seem to work.

On Thonny (IDE for uPython) I still see the boot text when doing a hard boot.

Also, Micropython seems to run after that, I have erased everything but left a basic micropython script that just prints a little text.

When I run esptool.py (even with --baud 115200) it always can read the parameters and ID of the device but it stops when reaching the "write_flash" part with a "write timeout".

Are there buttons I need to push? Both buttons on the Olimex boards seem to do nothing here.

I know there is a flashing tutorial in the Tasmota docs but that's what I tried to follow to no avail...

Thanks for any hints!
Andreas

**Comment by barbudor** [comment]
Did you tried to flash through the on board usb interface ?

I don't see any requirement for pressing buttons. The board includes the circuitry to drive EN and GPIO0 from the on-board usb/serial chip like most dev kit.
The only advice is to not connect simultaneously ethernet and us especially when using POE

**Reply by hagenbuch** [reply, marked as answer]
Yes, I connected to the Micro USB as usual. With Mac (Chrome) and with Linux (Chromium and terminal).

Yes and I didn't connect Ethernet... and as long as I develop, I have never PoE running.

Thanks for your reply anyway..

I also tried to flash a simple ESP32 example via Arduino IDE.. it worked and I see the output even in the logs of the web installer:

    rst:0x1 (POWERON_RESET),boot:0x1b (SPI_FAST_FLASH_BOOT)
    configsip: 0, SPIWP:0xee
    clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
    mode:DIO, clock div:1
    load:0x3fff0030,len:1184
    load:0x40078000,len:13260
    load:0x40080400,len:3028
    entry 0x400805e4
    ESP32 Chip model = ESP32-D0WDQ6 Rev 1
    This chip has 2 cores
    Chip ID: 10419204
    ESP32 Chip model = ESP32-D0WDQ6 Rev 1
    This chip has 2 cores
    Chip ID: 10419204`

but again, no cigar.

BUTTTTT!

Then I tried again to flash a tasmota32.factory.bin (Factory is important!) via esptool.py and I succeeded:

    $ esptool.py --port /dev/ttyUSB0 --baud 115200 write_flash 0x0 Downloads/tasmota32.factory.bin 
    esptool.py v3.1
    Serial port /dev/ttyUSB0
    Connecting....
    Detecting chip type... ESP32
    Chip is ESP32-D0WDQ6 (revision 1)
    Features: WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
    Crystal is 40MHz
    MAC: 24:0a:<secret>
    Uploading stub...
    Running stub...
    Stub running...
    Configuring flash size...
    Flash will be erased from 0x00000000 to 0x00229fff...
    Compressed 2268624 bytes to 1536981...
    Wrote 2268624 bytes (1536981 compressed) at 0x00000000 in 135.9 seconds (effective 133.6 kbit/s)... 
    Hash of data verified.
    Leaving...
    Hard resetting via RTS pin...

For those who're wondering: I used a version for ESP32 2 cores that has scripting and SDHC  compiled in.
